raiden/tests/integration/rpc/assumptions/test_nonce_assumptions.py

```diff
@@ -0,0 +1,23 @@
+import pytest
+
+from raiden.network.rpc.client import JSONRPCClient
+from raiden.tests.utils.smartcontracts import deploy_rpc_test_contract
+
+
+def test_transact_is_rejected_if_the_nonce_is_too_low(deploy_client: JSONRPCClient) -> None:
+    contract_proxy, _ = deploy_rpc_test_contract(deploy_client, "RpcTest")
+
+    # Create a new instance of the JSONRPCClient, this will store the current available nonce
+    client_invalid_nonce = JSONRPCClient(deploy_client.web3, deploy_client.privkey)
+
+    check_block = deploy_client.get_checking_block()
+    txhash = contract_proxy.transact("ret", contract_proxy.estimate_gas(check_block, "ret"))
+
+    # Wait for the transaction to be mined (cuncurrent transactions are tested
+    # by test_local_transaction_with_zero_gasprice_is_mined)
+    deploy_client.poll(txhash)
+
+    with pytest.raises(ValueError):
```

Comment:
Perhaps add a comment here that the `deploy_client` has the correct nonce since it was used to mine the transaction but the `client_invalid_nonce` is still in the old nonce and hence the nonce too low error happens?

Reply:
right above I have a comment for the converse:

> Create a new instance of the JSONRPCClient, this will store the current available nonce

isn't that sufficient?

Reply:
I saw that. But I did not get why the thing was supposed to fail later. I had to think a bit in order to understand what you were doing. Which is why I left the suggestion to spell it out more clearly.